### PR TITLE
Data Classification with Kernel SVM in Scikit-learn

### LINKED ISSUE #5188: Data classification with Kernel SVM in Scikit learn Library
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

### Topic Suggestion
Enter topic suggestion here...

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed article introduction
Kernel SVM is an unsupervised learning algorithm used in non-linear classification problems. When the data is linearly separable, we use SVM to obtain a classifier with an optimal margin. With the SVM, data can be perfectly classified using a linear hyperplane. This type of SVM is called Hard SVM. In other cases, the data might not be linearly separable, but still, we can classify them with minimum classification error. This type of SVM is known as the Soft SVM.

Although we can approach non-linear data using the soft SVM, it turns out that there are cases when the data is highly mixed up, and if we try to classify them using the soft SVM, we may end up obtaining a model whose accuracy is below 50%. In such situations, we use a kernelized SVM.

Typically, when dealing with non-linear data, one way to make the data linearly separable is by mapping it to higher dimensions. We then learn the existing relationships between features on the newly transformed data. The challenge with this approach is that it's time-consuming and requires high processing power. To overcome this challenge, we employ the kernel tricks, which utilize the original feature space, learn the underlined relationship in their higher dimensions. So the beauty of the kernel trick is that we don't need to map the data to higher dimensions to classify them optimally. This makes this approach to classification more efficient.

In this article, we will introduce the two types of linear SVM. After that, with the aid of a plot, we shall try to understand what we mean by referring to the data as non-linearly separable and how to map it to higher dimensions to make it linearly separable. Then, using an example, we shall then learn how the kernel trick works, and the kernel functions used when handling non-linear data. Finally, we shall understand the Gaussian kernel, and it's used with SVM to classify non-linear data.

#### Key takeaways 
The reader is expected to know;
- How does the SVM algorithm classify data classes.
- What's the kernel SVMs and in which scenario is applicable.
- How to transform non linearly separable datasets from low dimensional space to higher dimensional space to make them linearly separable
- How to define the kernel trick and how does it work.
- Various types of kernel functions are used with SVM.
- Most utilized kernel function when dealing with non-linear data
- How to implement the kernel SVM classifier on a dataset using Scikit-learn library in python

#### Article quality
This article will build up from the linear SVM from which we shall introduce the concept of the kernel SVM before we dive deeper into it; thus, the learner will have a firm understanding of the topic. Furthermore, all discussion will be to the point; hence, there will be no unnecessary explanation, making the content wordy and boring to fall along. Finally, we will visualize how our model classified our dataset, unlike many articles online, which only implement the model and leave it hugging without showing the learner how they can visualize classification classes.

#### References
N/A
### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by mbici**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5188

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing. @mbici
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by Kayere** [comment]
plagiarism check done.

File: content/articles/data-classification-with-kernel-svm-in-scikit-learn/index.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+![different](/engineering-education/kernel-svm-in-python/transformed-data.png)
+
+As we can see in the above plot, transforming the data is now linearly separable, and we can classify it using a linear hyperplane. Since we mapped our data from two to three dimensions, the decision boundary that optimally classifies our data is a plane.
+
+Most non-linear datasets are usually in high dimensional space in the real-world scenario. Mapping these features further to higher dimensional(kernel space) can result in even theoretically infinite-dimensional feature space. As a result, the processing time and power required on this kernel space are very high, making this approach inefficient. Also, the other problem associated with this approach is that the model obtained on such a higher dimension is very complex. Therefore, we employ a technique known as the *kernel trick* to make things less complex. This technique enables us to work in the input space without visiting the kernel space.
```
Comment (jumamiller) [inline content/articles/data-classification-with-kernel-svm-in-scikit-learn/index.md:39]: This is very hard to read!

Comment (jumamiller) [reply content/articles/data-classification-with-kernel-svm-in-scikit-learn/index.md:39]: You may break it into 2 or more paragraphs!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, is there a way we can format the vectors  such that they don't appear as normal texts?

**Comment by mbici** [comment]
Hello @jumamiller ,
The requested changes are in made.

**Comment by mbici** [comment]
Hello @jumamiller ,
I've made the requested changes

**Comment by LinusMuema** [comment]
Hi @mbici 

Really awesome article. Keep it up!

cc: @jumamiller

**Pull request merged**